### PR TITLE
chore: changelog for v4

### PR DESCRIPTION
**Pull request by vladfrangu**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+4.0.0 / 2023/11/07
+====================
+- **BREAKING**: This module is now ESM only.
+  - You will need to either migrate your projects to ESM, or import `got-scraping` in an async context via `await import('got-scraping');`
+- **BREAKING**: This module now requires Node.js >= 18
```
Comment (B4nan) [inline CHANGELOG.md:5]: wait a sec, why is that? our node 16 crawlee tests are passing with this, right?

Comment (vladfrangu) [reply CHANGELOG.md:5]: Oh tru, but nodejs 16 is also dead officially... I guess I could drop the requirement back to >= 16 for now

**Review by B4nan — CHANGES_REQUESTED** [review]
please update the readme too, it still uses CJS examples

File: README.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ $ npm install got-scraping
 ```
 
 **Note:**
-> - Node.js >=15.10.0 is required due to instability of HTTP/2 support in lower versions.
+> - Node.js >=16 is required due to instability of HTTP/2 support in lower versions.
```
Comment (B4nan) [inline README.md:12]: there should be (somewhere) a note about this project being ESM only too

**Review by B4nan — APPROVED** [review]

**Pull request merged**